### ISSUE TITLE
Unable to load kibana dashboards from zip file

### ISSUE THREAD

**Issue by ycombinator** [elastic/beats#12211]
Observed on `master` / `8.0.0`.

When using the `setup.dashboards.file` option, setting it to a zip file containing Kibana dashboard files, Packetbeat does not actually load the dashboards in the zip file into Kibana.

Relevant settings in `packetbeat.yml`:

```
setup.dashboards.file: /path/to/dashboards.zip
setup.kibana.host: "localhost:5601"
```

Note that if the zip file is extracted and the `setup.dashboards.directory` option, pointed to that folder, is used instead of the `setup.dashboards.file` option, the dashboards do get loaded into Kibana as expected.

**Comment by eedugon** [comment]
Zip example used when reproducing this:
[packetbeat_test.zip](https://github.com/elastic/beats/files/3227451/packetbeat_test.zip)

**Comment by dplavcic** [comment]
TLDR (importing Kibana dashboard from archive);

Libbeat implementation currently requires predefined directory structure to be able to import dashboards from archive
to Kibana.
Predefined folder structure is following:
    required structure: randomDirName/beatName/kibanaVersion/dashboard
    full working example: kibana/filebeat/7/dashboard

    not working example: kibana/7/dashboard

NOTE: I tested this with filebeat on commit c9613650f47458b449302db3edbcf08473c2071c

1. Current implementation of ImportArchive (beats/libbeat/dashboards/importer.go:194) assumes predefined directory
   structure to be available when importing dashboards from archive to Kibana. More precisely, it requires two levels of
   directories to be available as shown in following code snippet (224 - first level, 232 - second level):
   ...
   beats/libbeat/dashboards/importer.go:224
```
    ...
    224   	dirs, err := getDirectories(target)  //first level
    225   	if err != nil {
    226   		return err
    227   	}
    228   	if len(dirs) != 1 {
    229   		return fmt.Errorf("Too many directories under %s", target)
    230   	}
    231
    232   	dirs, err = getDirectories(dirs[0])  //second level
    233   	if err != nil {
    234   		return err
    235   	}
    ...
```

    After execution on line 232, 'dirs' variable will contain following values:
    dirs = {[]string} len: 1, cap: 1
      0 = {string} '/tmp/tmp803967903/kibana/filebeat'

    Following part is in charge of importing dashboards from '/tmp/tmp803967903/kibana/filebeat' directory to Kibana:
    ...
    beats/libbeat/dashboards/importer.go:237
   ```
 ...
    237	   for _, dir := range dirs {
    238	       imp.loader.statusMsg("Importing Kibana from %s", dir)
    239		       if imp.cfg.Beat == "" || filepath.Base(dir) == imp.cfg.Beat {
    240			   err = imp.ImportKibanaDir(dir)
    241			       if err != nil {
    242				       return err
    243			        }
    244	    	   }
    245 	}
    ...
```

    On line 239, we check if imp.cfg.Beat has a value or if second level directory name matches Beat name.
    If previous is true, then imp.ImportKibanaDir(dir) is called. Otherwise func returns nil (no warning is logged,
    no dashboards are imported).

    Previously described line, 239, is additional check that is not used when importing files from non archive
    directory, as we can see on 'beats/libbeat/dashboards/importer.go:85'

   ```
 ...
    78    func (imp Importer) Import() error {
    	  ...
    84            } else {
    85                err := imp.ImportKibanaDir(imp.cfg.Dir)
    86                if err != nil {
    87                    return errw.Wrapf(err, "Error importing directory %s", imp.cfg.Dir)
    88                }
    89            }
    90            return nil
    91       }
    ...
```

    Implementation of ImportKibanaDir (beats/libbeat/dashboards/importer.go:296) checks for another directory in
    hierarchy, and that is directory with name that matches Kibana version, in this case '7' (which is hardcoded):
   ```
 ...
    299    versionPath := "7"
    300
    301    dir = path.Join(dir, versionPath)
    302
    303    imp.loader.statusMsg("Importing directory %v", dir)
    304
    305    if _, err := os.Stat(dir); err != nil {
    306        return newErrNotFound("No directory %s", dir)
    307    }
    ...
```

    Up until now required directory structure is:
    '{firstLevel: kibana}/{secondLevel: beat name}/{thirdLevel: 7}/'
    value: dir = {string} "/tmp/tmp054631736/kibana/filebeat/7"

    After that, another predefined subdirectory is checked, in this case it is 'dashboard' subdirectory which is added
    in line (319).
    ...
    beats/libbeat/dashboards/importer.go:317
   ```
 ...
    317    	wantDashboards := false
    318       	if !imp.cfg.OnlyIndex {
    319       		check = append(check, "dashboard")
    320       		wantDashboards = true
    321       	}
    322
    323       	types := []string{}
    324       	for _, c := range check {
    325       		if imp.subdirExists(dir, c) {
    326       			types = append(types, c)
    327       		}
    328       	}
    ...
```

    All files on path 'kibana/filebeat/7/dashboard/' are imported to Kibana:
    1. /tmp/tmp054631736/kibana/filebeat/7/dashboard/Packetbeat-cassandra.json
    2. /tmp/tmp054631736/kibana/filebeat/7/dashboard/Packetbeat-flows.json
    3. ...


Proposal:
1. Update documentation where 'setup.dashboards.file' is mentioned to include required directory structure example.

2. Increase logging on beats/libbeat/dashboards/importer.go:239 or throw error if
   'imp.cfg.Beat == "" || filepath.Base(dir) == imp.cfg.Beat' is false to indicate that no Kibana dashboards will be
   created.
   NOTE: I will create PR for this if needed.

3. I currently don't know why there is additional check 'if imp.cfg.Beat == "" || filepath.Base(dir) == imp.cfg.Beat'
   in beats/libbeat/dashboards/importer.go:239, so I can't yet propose to upgrade/remove this.
   NOTE: I will create PR for this if needed.

4. Remove hardcoded Kibana version (there is already opened issue for this, but I can't find it now)
